lib/Unexpected.js

```diff
@@ -1126,15 +1126,22 @@ Unexpected.prototype._expect = function expect(subject, testDescriptionString) {
 
         if (!assertionRule) {
             var tokens = testDescriptionString.split(' ');
-            for (var n = tokens.length - 1; n > 0 ; n -= 1) {
+            OUTER: for (var n = tokens.length - 1; n > 0 ; n -= 1) {
                 var prefix = tokens.slice(0, n).join(' ');
-                var argsWithAssertionPrepended = [ tokens.slice(n).join(' ') ].concat(args);
+                var remainingTokens = tokens.slice(n);
+                var argsWithAssertionPrepended = [ remainingTokens.join(' ') ].concat(args);
                 assertionRule = that.lookupAssertionRule(subject, prefix, argsWithAssertionPrepended, true);
                 if (assertionRule) {
-                    // Great, found the longest prefix of the string that yielded a suitable assertion for the given subject and args
-                    testDescriptionString = prefix;
-                    args = argsWithAssertionPrepended;
-                    break;
+                    // Found the longest prefix of the string that yielded a suitable assertion for the given subject and args
+                    // To avoid bogus error messages when shifting later (#394) we require some prefix of the remaining tokens
+                    // to be a valid assertion name:
+                    for (var i = 1 ; i < remainingTokens.length ; i += 1) {
+                        if (remainingTokens.slice(0, i + 1).join(' ') in that.assertions) {
+                            testDescriptionString = prefix;
+                            args = argsWithAssertionPrepended;
+                            break OUTER;
```

Comment:
right, that makes sense. i'm guessing there's a reason why the outer loop is iterating in reverse?